Aptoide/appcoins-wallet-android
# [MOB-893] fingerprint payment flow

**What does this PR do?**

   Authentication is requested whenever a payment, transfer flow or top up is executed, if the authentication settings are on.

**Database changed?**

   No

**Where should the reviewer start?**

IabActivity.kt
TopUpActivity.kt
TransferFragment.kt

**How should this be manually tested?**

- Open Trivial Drive and try to buy gas with each different payment method;
- Make a Top Up;
- Transfer money to another wallet;
- Rotate the authentication prompt when repeating the above tests;
- Turn off authentication settings and check if the payments, transfers and top ups are still working properly.

**What are the relevant tickets?**

https://aptoide.atlassian.net/browse/MOB-893

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.)




**Code Review Checklist**
